File: backend/bottari/src/main/java/com/bottari/repository/TeamMemberRepository.java
package com.bottari.repository;

import com.bottari.teambottari.domain.TeamMember;
import java.util.List;
import org.springframework.data.jpa.repository.JpaRepository;

public interface TeamMemberRepository extends JpaRepository<TeamMember, Long> {

    List<TeamMember> findAllByTeamBottariId(final Long teamBottariId);
}

File: android/Bottari/core/domain/src/main/java/com/bottari/core/domain/repository/TeamMemberRepository.kt
package com.bottari.core.domain.repository

import com.bottari.core.domain.model.team.member.TeamMember
import com.bottari.core.domain.model.team.member.TeamMemberStatus
import com.bottari.core.domain.model.team.member.TeamStatus

interface TeamMemberRepository {
    suspend fun fetchTeamMembers(id: Long): Result<TeamStatus>

    suspend fun fetchTeamMembersStatus(id: Long): Result<List<TeamMemberStatus>>

    suspend fun sendRemindByMemberMessage(
        teamBottariId: Long,
        memberId: Long,
    ): Result<Unit>

    suspend fun joinTeamBottari(inviteCode: String): Result<Unit>

    suspend fun fetchTeamBottariMembers(teamBottariId: Long): Result<List<TeamMember>>
}
